### ISSUE TITLE
Using the browser's built-in print menu bypasses our pre-print actions

### ISSUE THREAD

**Issue by mgwalker** [18F/cms-hitech-apd#1354]
If the user clicks `File -> Print` in their browser, the app won't run its pre-print actions, which prevents us from doing things like expanding numbers to full size (#1261).  It all works if they use the print button in the sidebar, though.

Furthermore, because we don't catch the event for when a [print has been sent off](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onafterprint), we also don't switch the app _back_ from print mode to regular mode.

### This task is done when...
 - [ ] using the `File -> Print` menu options in the browser triggers pre-print activities
 - [ ] a print job being sent to the printer or canceled triggers post-print activities

The second half of this bug feels pretty critical to me.  Without fixing it, users who click the print button will have an app stuck in print mode until they reload the page.

**Comment by mgwalker** [comment]
Ugh, furthermore...

Because our app changes go through the app store and then React's re-rendering process, they happen asynchronously, but not before the next event loop.  The browser's print behavior happens on the same event loop as when they press the button, so our changes won't happen until after the content is gobbled up and formatted for printing.  The print event is non-cancellable (rightfully so!) so we can't delay it until after the app has had a chance to change over.

The bottom line is that we cannot rely on application state changes to handle printing issues.  It has to be done entirely in stylesheets.

**Comment by mgwalker** [comment]
So, I created this as a bug, but I think since we can't work around it, we can close it - it's not a bug, it's a limitation of the world in which we exist.
